types/src/database_sql.rs

```diff
@@ -1910,14 +1910,18 @@ impl Loader<BookmarkedOn> for Database {
     ) -> Result<HashMap<BookmarkedOn, Self::Value>, Self::Error> {
         let document_id: Vec<_> = keys.iter().map(|k| k.0).collect();
         let user_id: Vec<_> = keys.iter().map(|k| k.1).collect();
-        let items = query_file!("queries/get_bookmarked_on.sql", document_id[0], user_id[0])
-            .fetch_all(&self.client)
-            .await?;
+        let items = query_file!(
+            "queries/get_document_bookmarked_on.sql",
+            document_id[0],
+            user_id[0]
+        )
```

Comment:
Something feels really off here. If this accepts a vec of keys we shouldn't implement to fetch just the first key. I'm not sure we really need a dataloader for this either for what it's worth. A data loader is normally for batching big operations. This is a small operation. If your query can only handle looking up one document at a time, you aren't writing a dataloader, it's just a normal query!

Did you model this off another dataloader that does this `[0]` stuff?

That is definitely a bug waiting to happen. All I need to do is call this with more than one document and it'll just fetch the first one and say the second doesn't exist.

Reply:
There is only going to be one element in the vector since the query in the front-end only ever sends in one of each. I'm pretty sure this is currently only used in the AnnotatedDoc query which has no manual input, so it would never have more than one thing in there. But I can see that it could be a problem later if someone manually puts in more than 1. I kind of remember trying to do this without sending in a vector but it gave an error saying it had to be in a vector so I believe that's why I made it like this. I can look at this again tonight when I return.

Reply:
Yeah the `DataLoader` stuff will always want it to be a vector. Just make a simple function on `Database` in that big `impl` block with all the others. `DataLoaders` can be great, but they solve a very specific problem.

Reply:
@nole2701 are you planning on leaving this as a data loader or will you be making this a function on Database (lines 35-1468)?

Reply:
Actually, I just noticed [this related discussion](https://github.com/NEU-DSG/dailp-encoding/pull/358#discussion_r1407188581), so you really should make this a function on `Database` rather than a data loader imo